File: 206/206.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	cur := head
	var prev *ListNode

	for cur != nil {
		next := cur.Next
		cur.Next = prev
		prev = cur
		cur = next
	}
	return prev
}

func printLinkList(head *ListNode) {
	for head != nil {
		fmt.Print(head.Val, " -> ")
		head = head.Next
	}
	fmt.Println("nil")
}

func main() {
	head := &ListNode{1, &ListNode{2, &ListNode{3, &ListNode{4, &ListNode{5, nil}}}}}
	fmt.Println("orgin LinkList:")
	printLinkList(head)

	res := reverseList(head)
	fmt.Println("reverse LinkList:")
	printLinkList(res)
}

File: 206/206.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let curr = head
    let prev = null

    while(curr != null){
        let next = curr.next
        curr.next = prev
        prev = curr
        curr = next
    }
    return prev
};

function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}
function printLinkedList(head) {
    let cur = head;
    let res = [];
    while (cur !== null) {
        res.push(cur.val);
        cur = cur.next;
    }
    console.log(res.join(" -> ") + " -> null");
}

let head = new ListNode(1, new ListNode(2, new ListNode(3, new ListNode(4, new ListNode(5)))));
console.log("原始鏈表:");
printLinkedList(head);

let reversedHead = reverseList(head);
console.log("反轉後鏈表:");
printLinkedList(reversedHead);